HW06.Rmd

```diff
@@ -1,92 +1,134 @@
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2","MECP2") #added MECP2
+
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+```
+
+```{r t-test}
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+for(i in 1:length(genes_of_interest)){
+  x<-ens.mapped_GBM%>%
+    filter(gene_symbol==genes_of_interest[i])#filter TPM data for the particular gene #i
+  print(paste("t-teset between H3K mutated and WT samples for",genes_of_interest[i],"expression"))
+  print(t.test(x[1,2:24],x[25:46]))#columns 2:24 are H3K27M cell lines and columns 25:46 are WT cell lines
+}
+
+#print out the t-test results
+```
+
+
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+tidy<-GBM.genes.of.interest%>%
+  rename_with(~paste0(.,"-H3K27_muated"),2:24)%>% #added suffix so that after gather(), each observation would have sample type information
+  rename_with(~paste0(.,"-WT"),25:46)%>%
+  gather(key="cell_line",value="TPM",2:46)%>%
+  separate(cell_line,c("cell_line","sample_type"),sep="-")%>%
+  select(-cell_line) #cell line information is not necessary for plots
+
+#create a graphing function
+graphing<-function(x){
+tidy%>%
+  filter(gene_symbol==genes_of_interest[x])%>%
+  ggplot(aes(x=sample_type,y=TPM,fill=sample_type))+
+  geom_boxplot()+
+  theme_bw()+
+  labs(title=paste(genes_of_interest[x],"Expression in GBM models by H3K27 mutated or WT Status"),x="H3K27",y="Expression_(TPM)")+
+  theme(legend.position = "none")
+}
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+for(i in 1:length(genes_of_interest)){
+  print(graphing(i))
+}
+```
+
+
```

Comment:
You are a coding master! I'm actually so impressed by how far you got. This is a really good attempt. I think in the latter half of the assignment (which I couldn't even get to), your code is clear enough. I understand at each step what you're trying to do, and I feel confident that you are using pretty condensed strategy. 

For the initial component of the assignment, I think there are a few places you could've condensed further to streamline. But I don't think that your approach was overly verbose either. I just think the many differing steps made it slightly more challenging to read than the others.  For example, I think you could've made more use of the select function just to make it crystal clear near the end what you're binding together as opposed to denoting the selected columns through "[4]". 

All in all though, I feel like you understood what you were doing and had a clear step wise approach to get through each step. I couldn't even get as far as you did so 10 stars! 🥇